### DISCUSSION TITLE
Producing and consuming messages with super stream over SSL(TLS V1.3) takes more time

### DISCUSSION THREAD

**Discussion by AbyssWorld** [rabbitmq/rabbitmq-server#11429] [Questions]
Sent around 50k of messages to super stream without SSL which is taking around 15 seconds to publish.
Sending same 50k messages to super stream over SLL (TLS V1.3) is taking around 2 minutes.
Any quick suggestions to improve this time of publishing messages over SSL? Is this slowness happening with everyone?

Using rabbitmq stream client with version 0.15.0.
Server configuration with single node is as below
[rabbitmq.txt](https://github.com/user-attachments/files/15785341/rabbitmq.txt)

Versions:
Rabbitmq -  3.13.0
Erlang - 26.2.2

**Comment by acogoluegnes** [comment]
You can use [Stream PerfTest](https://github.com/rabbitmq/rabbitmq-stream-perf-test/) to run several workloads to find out more about the performance of your system. Use one stream at first (no super stream), no TLS, set the message size like your use case, then [use TLS](https://rabbitmq.github.io/rabbitmq-stream-perf-test/stable/htmlsingle/#connection).

Please share also your server configuration and the client library you are using.

**Reply by AbyssWorld** [reply]
Using rabbitmq stream client with version 0.15.0.
Server configuration with single node is as below
[rabbitmq.txt](https://github.com/user-attachments/files/15785341/rabbitmq.txt)
